docs/modules/ROOT/pages/spring-cloud-gateway-server-webflux/gatewayfilter-factories/addresponseheadersifnotpresent-factory.adoc

```diff
@@ -0,0 +1,44 @@
+[[addresponseheadersifnotpresent-gatewayfilter-factory]]
+= `AddResponseHeadersIfNotPresent` `GatewayFilter` Factory
+
+The `AddResponseHeadersIfNotPresent` `GatewayFilter` factory takes a collection of `name` and `value` pairs separated by colon.
+The following example configures an `AddResponseHeadersIfNotPresent` `GatewayFilter`:
+
+.application.yml
+[source,yaml]
+----
+spring:
+  cloud:
+    gateway:
+      routes:
+      - id: add_response_headers_if_not_present_route
+        uri: https://example.org
+        filters:
+        - AddResponseHeadersIfNotPresent=X-Response-Color-1:blue,X-Response-Color-2:green
+----
+
+
+This listing adds 2 headers `X-Response-Color-1:blue` and `X-Response-Color-2:green` to the downstream response's headers for all matching requests.
+This is similar to how `AddResponseHeader` works, but unlike `AddResponseHeader` it will do it only if the header is not already there.
```

Comment:
Wouldn't it be useful to also add the option to override the header if it is present?  The default could be to do nothing if the header is there but we could add a configuration option to allow the override behavior

cc: @spencergibb

Reply:
I take back my comment here, I think we should add the option to override the header or not to `AddResponseHeader`

Reply:
This is not resolved...

Reply:
We should not add 'ResponseHeadersIfNotPresentGatewayFilter'. Instead, we should modify 'ResponseHeaderGatewayFilter' to support the override option.

Is this what you mean?

Reply:
Yes exactly